lxd/storage/backend_lxd.go

```diff
@@ -1487,6 +1487,14 @@ func (b *lxdBackend) EnsureImage(fingerprint string, op *operations.Operation) e
 		return nil // Nothing to do for drivers that don't support optimized images volumes.
 	}
 
+	// We need to lock this operation to ensure that the image is not being
+	// created multiple times.
+	unlock := drivers.Lock(b.name, drivers.VolumeTypeImage, fingerprint)
+	if unlock == nil {
+		return nil
```

Comment:
@monstermunchkin this will be moot if you change drivers.Lock to never return nil, but in its current form, if there has been an issue locking then you should probably not return nil meaning no error.